src/collectors/cgroups.plugin/cgroup-network.c

```diff
@@ -611,15 +611,28 @@ void call_the_helper(pid_t pid, const char *cgroup) {
         nd_log(NDLS_COLLECTORS, NDLP_ERR, "cannot execute cgroup-network helper script: %s", command);
 }
 
+int ishex(char c) {
+    return (c >= '0' && c <= '9') ||
+           (c >= 'a' && c <= 'f') ||
+           (c >= 'A' && c <= 'F');
+}
+
+int is_valid_hex_escape(const char *arg) {
+    return (arg[0] == '\\') &&
+           (arg[1] == 'x') &&
+           ishex(arg[2]) &&
+           ishex(arg[3]);
+}
```

Comment:
should we make sure that arg is

- not NULL
- at least 4 characters

? cc @stelfrag

Reply:
Hello @intelfx ,

What @ilyam8 was suggesting is that, before testing `arg[0]`, it would be ideal to first verify whether the `arg` variable is a valid address.

Reply:
Hello @intelfx ,

I think it would be a good idea to add a check for `arg`. If, at some point, someone calls the function without verifying for NULL in their code, it could lead to potential issues.

Best regards!

Reply:
Thank you! I am going to test your PR now.